e2e/create-cluster_test.go

```diff
@@ -200,8 +200,35 @@ var _ = Describe("CreateCluster", func() {
 		Expect(kvCluster.Status.Ready).To(BeTrue())
 	}
 
+	setFakeAnnotationOnMachine := func(g Gomega) {
+		machineList := &infrav1.KubevirtMachineList{}
+		g.Expect(k8sclient.List(context.Background(), machineList, client.InNamespace(namespace))).To(Succeed())
+		g.Expect(machineList.Items).ToNot(BeEmpty(), "expecting a non-empty list of machines")
+
+		for _, machine := range machineList.Items {
+			patch := client.RawPatch(types.JSONPatchType, []byte(`[{"op":"add","path":"/metadata/annotations","value":{"fake-annotation":"to-trigger-reconcile"}}]`))
+			g.Expect(k8sclient.Patch(context.Background(), &machine, patch)).To(Succeed())
+
+		}
+	}
+
+	removeFakeAnnotationFromMachine := func(g Gomega) {
+		machineList := &infrav1.KubevirtMachineList{}
+		g.Expect(k8sclient.List(context.Background(), machineList, client.InNamespace(namespace))).To(Succeed())
+		g.Expect(machineList.Items).ToNot(BeEmpty(), "expecting a non-empty list of machines")
+
+		for _, machine := range machineList.Items {
+			patch := client.RawPatch(types.JSONPatchType, []byte(`[{"op":"remove","path":"/metadata/annotations/fake-annotation"}]`))
+			g.Expect(k8sclient.Patch(context.Background(), &machine, patch)).To(Succeed())
+
+		}
+	}
+
 	waitForMachineReadiness := func(numExpectedReady int, numExpectedNotReady int) {
 		Eventually(func(g Gomega) {
+			setFakeAnnotationOnMachine(g)
+			defer removeFakeAnnotationFromMachine(g)
+
```

Comment:
I can see how forcing the reconciliation improves the test time and consistency, but my concern is that it might mask real issues caused by the reconcile loop not popping when we need it to. 

Is it possible for us to test this without forcing the reconciliation? it's okay if the wait time needs to increase?